src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitor.scala

```diff
@@ -156,12 +156,11 @@ class SubmissionMonitor(workspaceName: WorkspaceName,
     }
 
   private def withWorkspaceContext[T](workspaceName: WorkspaceName, writeLock: Boolean, txn: RawlsTransaction)(op: (WorkspaceContext) => T ) = {
+    txn.withLock(workspaceName, writeLock) {
```

Comment:
Note that SubmissionMonitor is using the non-Future lock here, because it's already running in a separate thread, so it can afford to block.